ADR/####-slsa-source-provenance-verification.md

```diff
@@ -0,0 +1,130 @@
+- **Gradual adoption**: Teams can enable source provenance verification incrementally
+- **Complements ADR 49**: Where [ADR 49](0049-vsa-support.md) trusts external VSAs to verify externally-built artifacts for release, this ADR detects and reports source VSAs (generated by external systems) for Konflux-built artifacts
+
+### Negative
+
+- **Race condition risk**: Source VSAs should be generated contemporaneously with commits (typically via GitHub Actions or GitLab pipelines), but Konflux's verification task may execute before the external automation completes. This can cause initial verification failures when:
+  - A commit is pushed to a repository
+  - Konflux immediately triggers a build pipeline
+  - The source VSA generation (GitHub Action/GitLab pipeline) is still running
+  - The verification task attempts to read git notes that don't exist yet
+
+  **Mitigation**: The movable test pattern from ADR 48 significantly helps with this issue. Because source verification runs as an independent integration test scenario rather than an embedded build task, it can be easily re-run once the source VSA is available. This is much easier than if the verification were embedded in the build pipeline, which would require a complete rebuild.
+
+- **Additional task required**: New verification task must be implemented and maintained (poc in [konflux-ci/build-definitions#2867](https://github.com/konflux-ci/build-definitions/pull/2867))
+- **OCI storage overhead**: Source VSAs are duplicated (git notes + OCI registry)
+- **Complexity in Conforma**: Policies must understand attestation chains rather than direct attestations
+- **External dependencies**: Requires source repository owners to set up VSA generation automation
+
+### Implementation Requirements
+
+1. **New Tekton task**: Implement `verify-source` source provenance verification task that:
+   - Accepts a public key parameter (configured via IntegrationTestScenario)
+   - Fetches git notes from cloned repositories
+   - Verifies source VSA signatures against the provided public key
+   - Verifies source VSA contents (format, subject, etc.)
+   - Pushes VSAs to OCI registry
+   - Exposes pullspec in `*_ARTIFACT_OUTPUTS` results format
+
+2. **Conforma policy updates**: Update Enterprise Contract policies to:
+   - Detect source provenance attestation chains
+   - Verify build provenance attestations about source VSAs
+   - Read the public key from test attestation task parameters and verify it matches expected signer identities for source repositories
+   - Determine if source verification levels meet release requirements
+
+3. **Documentation for source repository owners**:
+   - How to generate source VSAs (GitHub Actions, GitLab CI templates)
+   - How to store VSAs as git notes
+   - What source verification levels mean
+
+4. **ITS Provisioning**: Update Konflux onboarding mechanisms to provision a source verification ITS automatically
```

Comment:
Are you proposing to enable this globally?

Reply:
yeah, sounds good. Should we make this explicit? Or modify the ADR to state minimum criteria that would be needed for this to be further used... 

But the global deployment of this also doesn't need to happen with Konflux-CI itself. It could be done for a specific deployment of Konflux? Organizations would need to have this lever if we they would require this on the policy verification.

Reply:
Updated the text to clarify that this is a capability for Konflux deployments to enable if they choose (e.g., for policy requirements), not a mandatory global rollout. Deployments wanting to adopt this can update their onboarding mechanisms accordingly.